web/src/stream_list.ts

```diff
@@ -281,6 +280,15 @@ export function rewire_stream_list_section_container_html(
     stream_list_section_container_html = value;
 }
 
+function get_section_channel_plus_icon_url(section: StreamListSection): string | undefined {
+    if (section.id === "normal-streams") {
+        return "#channels/new";
+    } else if (!["pinned-streams", "dormant-streams"].includes(section.id)) {
+        return `#channels/folders/${section.id}/new`;
+    }
+    return undefined;
```

Comment:
I feel like this would probably be nicer if `section` just had a `folder_id` field, that's null for non-folder sections, and then just wrote 
```
if (section.folder_id !== null) { 
    return `#channels/folders/${section.folder_id}/new`;
} else if (section.id="normal-streams") { 
    return "#channels/new"; 
}
return undefined;
```

So much more readable locally without having to know what the complete set of non-channel-folders are.